### PR TITLE
Allow to define the maximum body size to passive scan

### PR DESCRIPTION
**Pull request by psiinon**
Partly addresses #3813

Not sure how to add unit tests for this :/

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanThread.java [hunk 1]
```diff
@@ -179,6 +179,23 @@ public void run() {
                     try {
                         // Parse the record
                         HttpMessage msg = href.getHttpMessage();
+                        if (this.pscanOptions.getMaxBodySizeInBytesToScan() > 0
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanThread.java:182]: Worth adding `msg.isResponseFromTargetHost()`.

**Comment by thc202** [comment]
- Should the option be explicit in that it applies only to response bodies?
 - Could you update the user guide?

**Comment by psiinon** [comment]
Changed to check both request and response bodies.

**Comment by kingthorin** [comment]
`PausableScheduledThreadPoolExecutorUnitTest` seems to have failed again. I tried to "Re-run" via the web UI but Travis doesn't seem to have started again. You might have to kick it with an amended commit.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]
Thanks

**Pull request merged**